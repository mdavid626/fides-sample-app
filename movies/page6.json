{"page":6,"results":[{"adult":false,"backdrop_path":"https://image.tmdb.org/t/p/original/geYUecpFI2AonDLhjyK9zoVFcMv.jpg","genre_ids":[16,28,14],"id":810693,"original_language":"ja","original_title":"劇場版 呪術廻戦 0","overview":"Yuta Okkotsu is a nervous high school student who is suffering from a serious problem—his childhood friend Rika has turned into a curse and won't leave him alone. Since Rika is no ordinary curse, his plight is noticed by Satoru Gojo, a teacher at Jujutsu High, a school where fledgling exorcists learn how to combat curses. Gojo convinces Yuta to enroll, but can he learn enough in time to confront the curse that haunts him?","popularity":358.685,"poster_path":"https://image.tmdb.org/t/p/original/3pTwMUEavTzVOh6yLN0aEwR7uSy.jpg","release_date":"2021-12-24","title":"Jujutsu Kaisen 0","video":false,"vote_average":8.3,"vote_count":731},{"adult":false,"backdrop_path":"https://image.tmdb.org/t/p/original/qjGrUmKW78MCFG8PTLDBp67S27p.jpg","genre_ids":[16,28,12,14],"id":635302,"original_language":"ja","original_title":"劇場版「鬼滅の刃」無限列車編","overview":"Tanjirō Kamado, joined with Inosuke Hashibira, a boy raised by boars who wears a boar's head, and Zenitsu Agatsuma, a scared boy who reveals his true power when he sleeps, boards the Infinity Train on a new mission with the Fire Hashira, Kyōjurō Rengoku, to defeat a demon who has been tormenting the people and killing the demon slayers who oppose it!","popularity":365.394,"poster_path":"https://image.tmdb.org/t/p/original/h8Rb9gBr48ODIwYUttZNYeMWeUU.jpg","release_date":"2020-10-16","title":"Demon Slayer -Kimetsu no Yaiba- The Movie: Mugen Train","video":false,"vote_average":8.3,"vote_count":2864},{"adult":false,"backdrop_path":"https://image.tmdb.org/t/p/original/2iGUavwv86Hubv3V5Iq4qEQXDfE.jpg","genre_ids":[53,18,27],"id":848058,"original_language":"es","original_title":"Cerdita","overview":"A bullied overweight teenager sees a glimpse of hope when her tormentors are brutally abducted by a mesmerizing stranger.","popularity":349.757,"poster_path":"https://image.tmdb.org/t/p/original/8EIQAvJjXdbNDMmBtHtgGqbc09V.jpg","release_date":"2022-10-07","title":"Piggy","video":false,"vote_average":6.7,"vote_count":245},{"adult":false,"backdrop_path":"https://image.tmdb.org/t/p/original/6RCf9jzKxyjblYV4CseayK6bcJo.jpg","genre_ids":[18],"id":804095,"original_language":"en","original_title":"The Fabelmans","overview":"Growing up in post-World War II era Arizona, young Sammy Fabelman aspires to become a filmmaker as he reaches adolescence, but soon discovers a shattering family secret and explores how the power of films can help him see the truth.","popularity":282.568,"poster_path":"https://image.tmdb.org/t/p/original/xId9zoiv5WPQOuxqykUDrlpmrUz.jpg","release_date":"2022-11-11","title":"The Fabelmans","video":false,"vote_average":8,"vote_count":295},{"adult":false,"backdrop_path":"https://image.tmdb.org/t/p/original/sCOHkah9RbFeZfFnfBrcykKCMNa.jpg","genre_ids":[27,878],"id":846778,"original_language":"en","original_title":"Margaux","overview":"As a group of seniors celebrate their final college days at a smart house, the house's highly advanced AI system, Margaux, begins to take on a deadly presence of her own. A carefree weekend of partying turns into a dystopian nightmare as they realize Margaux's plans to eliminate her tenants one way or another. Time begins to run out as the group desperately tries to survive and outsmart the smart home.","popularity":340.409,"poster_path":"https://image.tmdb.org/t/p/original/uNzgeMetu9l4q9NDw7gtiUFwPOJ.jpg","release_date":"2022-09-09","title":"Margaux","video":false,"vote_average":6.8,"vote_count":59},{"adult":false,"backdrop_path":"https://image.tmdb.org/t/p/original/fOy2Jurz9k6RnJnMUMRDAgBwru2.jpg","genre_ids":[16,10751,35,14],"id":508947,"original_language":"en","original_title":"Turning Red","overview":"Thirteen-year-old Mei is experiencing the awkwardness of being a teenager with a twist – when she gets too excited, she transforms into a giant red panda.","popularity":297.702,"poster_path":"https://image.tmdb.org/t/p/original/qsdjk9oAKSQMWs0Vt5Pyfh6O4GZ.jpg","release_date":"2022-03-10","title":"Turning Red","video":false,"vote_average":7.5,"vote_count":3696},{"adult":false,"backdrop_path":"https://image.tmdb.org/t/p/original/h2VmRSlnuH8RCPa7loYZK350q6J.jpg","genre_ids":[27,53],"id":929562,"original_language":"en","original_title":"In the Forest","overview":"Helen and her daughter Emily reluctantly accompany Helen’s father Stan on a family camping trip, driving deep into the forest for a real outdoors experience, only to have angry landowner Howard arrive and force them to leave. When their RV gets stuck, Helen decides to seek help but discovers a disturbing secret on the man’s property. She must then fight to protect her family and make it out of the forest alive before being hunted down.","popularity":331.801,"poster_path":"https://image.tmdb.org/t/p/original/pWKTaGN9WMcKHKAir9POsKm4vEo.jpg","release_date":"2022-01-28","title":"In the Forest","video":false,"vote_average":6.1,"vote_count":13},{"adult":false,"backdrop_path":"https://image.tmdb.org/t/p/original/5jkE2SzR5uR2egEb1rRhF22JyWN.jpg","genre_ids":[12,14],"id":671,"original_language":"en","original_title":"Harry Potter and the Philosopher's Stone","overview":"Harry Potter has lived under the stairs at his aunt and uncle's house his whole life. But on his 11th birthday, he learns he's a powerful wizard—with a place waiting for him at the Hogwarts School of Witchcraft and Wizardry. As he learns to harness his newfound powers with the help of the school's kindly headmaster, Harry uncovers the truth about his parents' deaths—and about the villain who's to blame.","popularity":278.767,"poster_path":"https://image.tmdb.org/t/p/original/wuMc08IPKEatf9rnMNXvIDxqP4W.jpg","release_date":"2001-11-16","title":"Harry Potter and the Philosopher's Stone","video":false,"vote_average":7.9,"vote_count":23785},{"adult":false,"backdrop_path":"https://image.tmdb.org/t/p/original/aIkG2V4UXrfkxMdJZmq30xO0QQr.jpg","genre_ids":[878,10751,28,12],"id":791155,"original_language":"en","original_title":"Secret Headquarters","overview":"While hanging out after school, Charlie and his friends discover the headquarters of the world’s most powerful superhero hidden beneath his home. When villains attack, they must team up to defend the headquarters and save the world.","popularity":349.654,"poster_path":"https://image.tmdb.org/t/p/original/8PsHogUfvjWPGdWAI5uslDhHDx7.jpg","release_date":"2022-08-12","title":"Secret Headquarters","video":false,"vote_average":6.8,"vote_count":189},{"adult":false,"backdrop_path":"https://image.tmdb.org/t/p/original/2DbSpBgjHASSOXe5eHC9qVtif0F.jpg","genre_ids":[10749,18],"id":452251,"original_language":"ko","original_title":"미용실 : 특별한 서비스","overview":"A legendary beauty salon that is sure to become a regular once you go!”Sir, would you like to have a very hot service?”In order to overcome the financial difficulties of the beauty parlor,It begins to attract the male guests with the special technique that makes it from head to toe.Kurt, Firm, and ‘it’ to open the hair salon is good!","popularity":306.478,"poster_path":"https://image.tmdb.org/t/p/original/z3Vse8jRnmKJyZFK6VSw1G6jHnY.jpg","release_date":"2016-12-29","title":"Beauty Salon: Special Service","video":false,"vote_average":5.6,"vote_count":8},{"adult":false,"backdrop_path":"https://image.tmdb.org/t/p/original/mDfJG3LC3Dqb67AZ52x3Z0jU0uB.jpg","genre_ids":[12,28,878],"id":299536,"original_language":"en","original_title":"Avengers: Infinity War","overview":"As the Avengers and their allies have continued to protect the world from threats too large for any one hero to handle, a new danger has emerged from the cosmic shadows: Thanos. A despot of intergalactic infamy, his goal is to collect all six Infinity Stones, artifacts of unimaginable power, and use them to inflict his twisted will on all of reality. Everything the Avengers have fought for has led up to this moment - the fate of Earth and existence itself has never been more uncertain.","popularity":274.657,"poster_path":"https://image.tmdb.org/t/p/original/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg","release_date":"2018-04-25","title":"Avengers: Infinity War","video":false,"vote_average":8.3,"vote_count":25995},{"adult":false,"backdrop_path":"https://image.tmdb.org/t/p/original/mMA2YNddowV8MZtxpbn0a7Yilum.jpg","genre_ids":[10752,36,28,18],"id":928123,"original_language":"zh","original_title":"长津湖之水门桥","overview":"In the follow-up to \"The Battle At Lake Changjin\", brothers Wu Qianli and Wu Wanli undertake a new task for the People's Volunteer Army, defending a bridge part of the American troops' escape route from the advancing Chinese.","popularity":304.05,"poster_path":"https://image.tmdb.org/t/p/original/ugiL6wIhl1OfPyv1gqLkTe45jLl.jpg","release_date":"2022-02-01","title":"The Battle at Lake Changjin: Water Gate Bridge","video":false,"vote_average":6.9,"vote_count":31},{"adult":false,"backdrop_path":"https://image.tmdb.org/t/p/original/hYZ4a0JvPETdfgJJ9ZzyFufq8KQ.jpg","genre_ids":[28,18,878],"id":629176,"original_language":"en","original_title":"Samaritan","overview":"Thirteen year old Sam Cleary  suspects that his mysteriously reclusive neighbor Mr. Smith is actually the legendary vigilante Samaritan, who was reported dead 20 years ago. With crime on the rise and the city on the brink of chaos, Sam makes it his mission to coax his neighbor out of hiding to save the city from ruin.","popularity":322.288,"poster_path":"https://image.tmdb.org/t/p/original/vwq5iboxYoaSpOmEQrhq9tHicq7.jpg","release_date":"2022-08-25","title":"Samaritan","video":false,"vote_average":6.9,"vote_count":1481},{"adult":false,"backdrop_path":"https://image.tmdb.org/t/p/original/hNzrnsH9FMMfITu2xQqaf70CRv5.jpg","genre_ids":[9648,14,53],"id":856245,"original_language":"es","original_title":"Matar a la Bestia","overview":"Emilia arrives at her Aunt Inés' hostel located on the Argentina-Brazil border, looking for her missing brother. In this lush jungle a dangerous beast which takes the form of different animals seems to be roaming around.","popularity":340.712,"poster_path":"https://image.tmdb.org/t/p/original/hAdkgE8lHelIQWpgrHk4wjEnbxQ.jpg","release_date":"2022-04-28","title":"To Kill the Beast","video":false,"vote_average":5,"vote_count":69},{"adult":false,"backdrop_path":"https://image.tmdb.org/t/p/original/atmII0hn3iQe3IWMBmIb3cc8EJZ.jpg","genre_ids":[18,27,10749],"id":791177,"original_language":"en","original_title":"Bones and All","overview":"Abandoned by her father, a young woman embarks on a thousand-mile odyssey through the backroads of America where she meets a disenfranchised drifter. But despite their best efforts, all roads lead back to their terrifying pasts and to a final stand that will determine whether their love can survive their otherness.","popularity":307.578,"poster_path":"https://image.tmdb.org/t/p/original/gH6ppXl2gd1VMKoccKbLK69ZHQq.jpg","release_date":"2022-11-18","title":"Bones and All","video":false,"vote_average":7.5,"vote_count":451},{"adult":false,"backdrop_path":"https://image.tmdb.org/t/p/original/AfvIjhDu9p64jKcmohS4hsPG95Q.jpg","genre_ids":[27,53],"id":756999,"original_language":"en","original_title":"The Black Phone","overview":"Finney Blake, a shy but clever 13-year-old boy, is abducted by a sadistic killer and trapped in a soundproof basement where screaming is of little use. When a disconnected phone on the wall begins to ring, Finney discovers that he can hear the voices of the killer’s previous victims. And they are dead set on making sure that what happened to them doesn’t happen to Finney.","popularity":271.839,"poster_path":"https://image.tmdb.org/t/p/original/lr11mCT85T1JanlgjMuhs9nMht4.jpg","release_date":"2022-06-22","title":"The Black Phone","video":false,"vote_average":7.8,"vote_count":3312},{"adult":false,"backdrop_path":"https://image.tmdb.org/t/p/original/729a5iDNxr5l2Zbc8lcwndlJavN.jpg","genre_ids":[878,27],"id":844396,"original_language":"en","original_title":"The Friendship Game","overview":"A group of teens in a small town come across a strange object that tests their loyalties to each other with increasingly destructive consequences the deeper into the game they go.","popularity":339.782,"poster_path":"https://image.tmdb.org/t/p/original/qXaJkaVTNMhrOylFi6xrWnBQGE8.jpg","release_date":"2022-11-11","title":"The Friendship Game","video":false,"vote_average":5.5,"vote_count":22},{"adult":false,"backdrop_path":null,"genre_ids":[],"id":1030419,"original_language":"ja","original_title":"銀河英雄伝説 Die Neue These 策謀 1","overview":"","popularity":331.355,"poster_path":"https://image.tmdb.org/t/p/original/brrgSFtcymZWaXl5a23GJRWdOSY.jpg","release_date":"2022-09-30","title":"Legend of the Galactic Heroes: Die Neue These - Intrigue 1","video":false,"vote_average":5.2,"vote_count":11},{"adult":false,"backdrop_path":"https://image.tmdb.org/t/p/original/3G1Q5xF40HkUBJXxt2DQgQzKTp5.jpg","genre_ids":[16,35,10751,14],"id":568124,"original_language":"en","original_title":"Encanto","overview":"The tale of an extraordinary family, the Madrigals, who live hidden in the mountains of Colombia, in a magical house, in a vibrant town, in a wondrous, charmed place called an Encanto. The magic of the Encanto has blessed every child in the family—every child except one, Mirabel. But when she discovers that the magic surrounding the Encanto is in danger, Mirabel decides that she, the only ordinary Madrigal, might just be her exceptional family's last hope.","popularity":266.251,"poster_path":"https://image.tmdb.org/t/p/original/4j0PNHkMr5ax3IA8tjtxcmPU3QT.jpg","release_date":"2021-10-13","title":"Encanto","video":false,"vote_average":7.7,"vote_count":7743},{"adult":false,"backdrop_path":"https://image.tmdb.org/t/p/original/qaTzVAW1u16WFNsepjCrilBuInc.jpg","genre_ids":[16,28,10751,35,878],"id":539681,"original_language":"en","original_title":"DC League of Super-Pets","overview":"When Superman and the rest of the Justice League are kidnapped, Krypto the Super-Dog must convince a rag-tag shelter pack - Ace the hound, PB the potbellied pig, Merton the turtle and Chip the squirrel - to master their own newfound powers and help him rescue the superheroes.","popularity":269.456,"poster_path":"https://image.tmdb.org/t/p/original/r7XifzvtezNt31ypvsmb6Oqxw49.jpg","release_date":"2022-07-27","title":"DC League of Super-Pets","video":false,"vote_average":7.4,"vote_count":1027}],"total_pages":36555,"total_results":731099}