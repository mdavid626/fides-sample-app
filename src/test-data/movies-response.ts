import { MoviesResponse } from '../../types/movies-response';
import { movies } from './movies';

export const moviesResponse1: MoviesResponse = {
  page: 1,
  results: movies,
  total_pages: 7,
};
