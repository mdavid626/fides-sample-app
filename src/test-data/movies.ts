import { Movie } from '../types/movies-response';

export const movie1: Movie = {
  adult: false,
  backdrop_path:
    'https://image.tmdb.org/t/p/original/s16H6tpK2utvwDtzZ8Qy4qm5Emw.jpg',
  genre_ids: [878, 12, 28],
  id: 76600,
  original_language: 'en',
  original_title: 'Avatar: The Way of Water',
  overview:
    'Set more than a decade after the events of the first film, learn the story of the Sully family (Jake, Neytiri, and their kids), the trouble that follows them, the lengths they go to keep each other safe, the battles they fight to stay alive, and the tragedies they endure.',
  popularity: 6516.443,
  poster_path:
    'https://image.tmdb.org/t/p/original/t6HIqrRAclMCA60NsSmeqe9RmNV.jpg',
  release_date: '2022-12-14',
  title: 'Avatar: The Way of Water',
  video: false,
  vote_average: 7.7,
  vote_count: 3343,
};

export const movie2: Movie = {
  adult: false,
  backdrop_path:
    'https://image.tmdb.org/t/p/original/dKqa850uvbNSCaQCV4Im1XlzEtQ.jpg',
  genre_ids: [9648, 53, 35],
  id: 661374,
  original_language: 'en',
  original_title: 'Glass Onion: A Knives Out Mystery',
  overview:
    'World-famous detective Benoit Blanc heads to Greece to peel back the layers of a mystery surrounding a tech billionaire and his eclectic crew of friends.',
  popularity: 6801.729,
  poster_path:
    'https://image.tmdb.org/t/p/original/vDGr1YdrlfbU9wxTOdpf3zChmv9.jpg',
  release_date: '2022-11-23',
  title: 'Glass Onion: A Knives Out Mystery',
  video: false,
  vote_average: 7.1,
  vote_count: 1868,
};

export const movie3: Movie = {
  adult: false,
  backdrop_path:
    'https://image.tmdb.org/t/p/original/6mEYUYdkKoVCMeYf3rTFj0L1uyv.jpg',
  genre_ids: [28, 35, 80, 53],
  id: 899112,
  original_language: 'en',
  original_title: 'Violent Night',
  overview:
    'When a team of mercenaries breaks into a wealthy family compound on Christmas Eve, taking everyone inside hostage, the team isn’t prepared for a surprise combatant: Santa Claus is on the grounds, and he’s about to show why this Nick is no saint.',
  popularity: 5267.458,
  poster_path:
    'https://image.tmdb.org/t/p/original/1XSYOP0JjjyMz1irihvWywro82r.jpg',
  release_date: '2022-11-30',
  title: 'Violent Night',
  video: false,
  vote_average: 7.7,
  vote_count: 715,
};

export const movies = [movie1, movie2, movie3];
