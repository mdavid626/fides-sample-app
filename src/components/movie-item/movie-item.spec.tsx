import React from 'react';
import { cleanup, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import {
  useAddToFavourites,
  useRemoveFromFavourites,
} from '../../hooks/movies-hooks/movies-hooks';
import { movie1 } from '../../test-data/movies';
import MovieItem from './movie-item';

jest.mock('../../hooks/movies-hooks/movies-hooks');

describe('movie-item', () => {
  beforeEach(() => {
    (useAddToFavourites as jest.Mock).mockReturnValue([jest.fn(), false]);
    (useRemoveFromFavourites as jest.Mock).mockReturnValue([jest.fn(), false]);
  });
  afterEach(cleanup);
  afterEach(jest.resetAllMocks);

  it('should render', () => {
    const { asFragment } = render(<MovieItem movie={movie1} />);
    expect(asFragment()).toMatchSnapshot();
    expect(useAddToFavourites).toHaveBeenCalledWith();
    expect(useRemoveFromFavourites).toHaveBeenCalledWith();
  });

  it('should render remove from favourites button', () => {
    render(<MovieItem movie={movie1} isFavourite />);
    expect(screen.queryByTitle('Add to favourites')).toBe(null);
    expect(screen.getByTitle('Remove from favourites')).toBeVisible();
  });

  it('should add custom className', () => {
    render(<MovieItem movie={movie1} className="myClassName" />);
    expect(screen.getByTestId('MovieItem-76600')).toHaveClass('MovieItem');
    expect(screen.getByTestId('MovieItem-76600')).toHaveClass('myClassName');
  });

  it('should be able to add to favourites', () => {
    const addToFavourites = jest.fn();
    (useAddToFavourites as jest.Mock).mockReturnValue([addToFavourites, false]);
    render(<MovieItem movie={movie1} />);
    userEvent.click(screen.getByTitle('Add to favourites'));
    expect(addToFavourites).toHaveBeenCalledWith(movie1);
  });

  it('should not be able to add to favourites when adding is in progress', () => {
    const addToFavourites = jest.fn();
    (useAddToFavourites as jest.Mock).mockReturnValue([addToFavourites, true]);
    render(<MovieItem movie={movie1} />);
    userEvent.click(screen.getByTitle('Add to favourites'));
    expect(addToFavourites).not.toHaveBeenCalled();
  });

  it('should be able to remove from favourites', () => {
    const removeFromFavourites = jest.fn();
    (useRemoveFromFavourites as jest.Mock).mockReturnValue([
      removeFromFavourites,
      false,
    ]);
    render(<MovieItem movie={movie1} isFavourite />);
    userEvent.click(screen.getByTitle('Remove from favourites'));
    expect(removeFromFavourites).toHaveBeenCalledWith(movie1);
  });

  it('should not be able to remove from favourites when remove is in progress', () => {
    const removeFromFavourites = jest.fn();
    (useRemoveFromFavourites as jest.Mock).mockReturnValue([
      removeFromFavourites,
      true,
    ]);
    render(<MovieItem movie={movie1} isFavourite />);
    userEvent.click(screen.getByTitle('Remove from favourites'));
    expect(removeFromFavourites).not.toHaveBeenCalled();
  });
});
