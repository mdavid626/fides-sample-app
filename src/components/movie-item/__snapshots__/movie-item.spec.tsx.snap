// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movie-item should render 1`] = `
<DocumentFragment>
  <div
    class="MovieItem"
    data-testid="MovieItem"
  >
    <img
      alt="movie poster"
      class="MovieItem-poster"
      src="https://image.tmdb.org/t/p/original/t6HIqrRAclMCA60NsSmeqe9RmNV.jpg"
    />
    <div
      class="MovieItem-titleAndOverview"
    >
      <div
        class="MovieItem-title"
      >
        Avatar: The Way of Water
      </div>
      <div
        class="MovieItem-overview"
      >
        Set more than a decade after the events of the first film, learn the story of the Sully family (Jake, Neytiri, and their kids), the trouble that follows them, the lengths they go to keep each other safe, the battles they fight to stay alive, and the tragedies they endure.
      </div>
    </div>
    <div
      class="MovieItem-dateAndFavourite"
    >
      <div>
        14.12.2022
      </div>
      <img
        alt="Add to favourites"
        class="MovieItem-favourite"
        src="star.svg"
        title="Add to favourites"
      />
    </div>
  </div>
</DocumentFragment>
`;
